components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticatorTest.java

```diff
@@ -0,0 +1,350 @@
+                "http://localhost:8080/playground2/oauth2client",
+                "Callback URL is not valid.");
+    }
+
+    @Test(priority = 1)
+    public void testGetUserInfoEndpoint() throws Exception {
+        Assert.assertEquals(openIDConnectAuthenticator.getUserInfoEndpoint(token, authenticatorProperties),
+                "https://localhost:9443/oauth2/userinfo", "unable to get the user infor endpoint");
+    }
+
+    @Test(priority = 2)
+    public void testGetSubjectAttributes() throws Exception {
+        Map<ClaimMapping, String> result;
+        mockOAuthClientResponse = mock(OAuthClientResponse.class);
+        when(mockOAuthClientResponse.getParam(OIDCAuthenticatorConstants.ACCESS_TOKEN)).thenReturn("eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9");
+        result = openIDConnectAuthenticator.getSubjectAttributes(mockOAuthClientResponse, authenticatorProperties);
+        Assert.assertTrue(result.isEmpty(), "result is not Empty.");
+    }
+
+    @Test
+    public void testInitiateAuthenticationRequest() throws Exception {
+        mockServletResponse = mock(HttpServletResponse.class);
+        mockServletRequest = mock(HttpServletRequest.class);
+        mockAuthenticationContext = mock(AuthenticationContext.class);
+        mockAuthenticationRequestContext(mockAuthenticationContext);
+    }
+
+    @Test
+    public void testPassProcessAuthenticationResponse() throws Exception {
+    }
+
+    @Test
+    public void testBuildClaimMappings() throws Exception {
+    }
+
+    @Test(priority = 4)
+    public void testGetContextIdentifier() throws Exception {
+        mockServletRequest = mock(HttpServletRequest.class);
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_GRANT_TYPE_CODE)).thenReturn("openid");
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_PARAM_STATE)).thenReturn("active,OIDC");
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.LOGIN_TYPE)).thenReturn("BASIC");
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_ERROR)).thenReturn("Error Login.");
+        Assert.assertNotNull(openIDConnectAuthenticator.getContextIdentifier(mockServletRequest), "Invalid context identifier.");
+
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_GRANT_TYPE_CODE)).thenReturn(null);
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_PARAM_STATE)).thenReturn("active,OIDC");
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.LOGIN_TYPE)).thenReturn(null);
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_ERROR)).thenReturn("Error Login.");
+        Assert.assertNotNull(openIDConnectAuthenticator.getContextIdentifier(mockServletRequest), "Invalid context identifier.");
+
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_GRANT_TYPE_CODE)).thenReturn(null);
```

Comment:
Let's simplify this
`when(mockServletRequest.getParameter(anyString())).thenReturn(null);`